consensus/scp/tests/mock_network/metamesh_topology.rs

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2018-2020 MobileCoin Inc.
+
+// "Metamesh" network topologies.
+
+// A metamesh consists of a set of (n) "organizations", each comprising (m) servers.
```

Comment:
@mfaulk can you elaborate on this?

I think we should be testing as many topologies as possible to help uncover bugs that are rare in real world testing. The SCP crate tests have been too easy for a long time. As long as there is quorum intersection, scp is supposed to guarantee finality.

Reply:
As an example, a 3k2-5k4 metamesh requires almost 15 seconds to externalize a single value, submitted to all nodes in parallel:
```
2020-07-01 02:54:31.1474 ...thread results: mm1-3,43,1, ...   <--- node_id, total messages broadcast, slots used
2020-07-01 02:54:31.1475 ...thread results: mm1-0,43,1, ...
2020-07-01 02:54:31.1475 ...thread results: mm2-4,42,1, ...
2020-07-01 02:54:31.1476 ...thread results: mm2-1,42,1, ...
2020-07-01 02:54:31.1476 ...thread results: mm0-4,42,1, ...
2020-07-01 02:54:31.1476 ...thread results: mm1-4,43,1, ...
2020-07-01 02:54:31.1477 ...thread results: mm0-1,43,1, ...
2020-07-01 02:54:31.1478 ...thread results: mm0-0,43,1, ...
2020-07-01 02:54:31.1479 ...thread results: mm1-1,43,1, ...
2020-07-01 02:54:31.1480 ...thread results: mm0-2,43,1, ...
2020-07-01 02:54:31.1526 ...thread results: mm2-3,42,1, ...
2020-07-01 02:54:31.1527 ...thread results: mm1-2,43,1, ...
2020-07-01 02:54:31.1527 ...thread results: mm0-3,43,1, ...
2020-07-01 02:54:31.1535 ...thread results: mm2-0,42,1, ...
2020-07-01 02:54:31.1553 ...thread results: mm2-2,42,1, ...
2020-07-01 02:54:31.1555 ...test results: 3k2-5k4,15055,1,20000,100,100, ...  <--- net name, runtime msec, #pushed, pushed/sec, vals/slot, scp timebase
2020-07-01 02:54:31.1556 ...test completed for 3k2-5k4: 15.055460568s (avg 0 tx/s), ...
```
I believe this topology has quorum intersection. I don't see how catchup is relevant because we use only one slot.

Reply:
I think this topology is a valid SCP network and is fine to test with. The issue Eran mentioned is in this structure, nodes within an organization are unable to catch up if they all fall behind at the same time. The catch-up rules look a lot like "If all the nodes of a quorum except me are ahead, then do catch-up," which can't be fulfilled if all the nodes in one organization fall behind at the same time.

An alternative structure is to make one node within an organization the "primary node", and configure other nodes within the organization as "secondary nodes" that include the primary in their quorum sets. Then, the primary can catch up from the network, and the secondaries catch up from the primary.